pkg/controller/operandrequest/handler.go

```diff
@@ -0,0 +1,126 @@
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	utilyaml "github.com/ghodss/yaml"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	odlmv1alpha1 "github.com/IBM/operand-deployment-lifecycle-manager/api/v1alpha1"
+)
+
+// Mutator is the struct of webhook
+// +k8s:deepcopy-gen=false
+type Mutator struct {
+	Reader  client.Reader
+	decoder *admission.Decoder
+}
+
+type csMaps struct {
+	NsMappingList []nsMapping `json:"namespaceMapping"`
+	DefaultCsNs   string      `json:"defaultCsNs"`
+}
+
+type nsMapping struct {
+	RequestNS string `json:"requested-from-namespace"`
+	CsNs      string `json:"map-common-service-namespace"`
+}
+
+// Handle mutates every creating pods
+func (p *Mutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+
+	klog.Infof("Webhook is invoked by OperandRequest %s/%s", req.AdmissionRequest.Namespace, req.AdmissionRequest.Name)
+	opreq := &odlmv1alpha1.OperandRequest{}
+	ns := req.AdmissionRequest.Namespace
+	err := p.decoder.Decode(req, opreq)
+	if err != nil {
+		klog.Error(err, "Error occurred decoding OperandRequest")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	copy := opreq.DeepCopy()
+
+	err = p.mutatePodsFn(ctx, copy, ns)
+
+	if err != nil {
+		klog.Error(err, "Error occurred mutating OperandRequest")
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	marshaledOpreq, err := json.Marshal(opreq)
+	marshaledcopy, err := json.Marshal(copy)
+
+	// admission.PatchResponse generates a Response containing patches.
+	return admission.PatchResponseFromRaw(marshaledOpreq, marshaledcopy)
+
+}
+
+// Mutates function values
+func (p *Mutator) mutatePodsFn(ctx context.Context, opreq *odlmv1alpha1.OperandRequest, namespace string) error {
+
+	csConfigmap := &corev1.ConfigMap{}
+
+	err := p.Reader.Get(ctx, types.NamespacedName{Namespace: "kube-public", Name: "common-service-maps"}, csConfigmap)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Infof("common service configmap kube-public/common-service-maps is not found: %v", err)
+			return nil
+		}
+		return fmt.Errorf("failed to fetch configmap kube-public/common-service-maps: %v", err)
+	}
+
+	commonServiceMaps := csConfigmap.Data["common-service-maps.yaml"]
+	var cmData csMaps
+	if err := utilyaml.Unmarshal([]byte(commonServiceMaps), &cmData); err != nil {
+		return err
+	}
+
+	for _, nsMapping := range cmData.NsMappingList {
+		if nsMapping.RequestNS == opreq.Namespace {
+			var defaultCsNs string
+			if cmData.DefaultCsNs == "" {
+				defaultCsNs = "ibm-common-services"
+			} else {
+				defaultCsNs = cmData.DefaultCsNs
+			}
+			for index, req := range opreq.Spec.Requests {
+				if req.RegistryNamespace == defaultCsNs {
+					req.RegistryNamespace = nsMapping.CsNs
+				}
+				opreq.Spec.Requests[index] = req
```

Comment:
Learned that `req` is not a pointer, `range` only copies the value from slice, we need to re-assign the value back to slice 👍